src/routes/device-logs.ts

```diff
@@ -216,6 +221,13 @@ export const store: RequestHandler = async (req: Request, res: Response) => {
 		const logs: DeviceLog[] = supervisor.convertLogs(ctx, body);
 		if (logs.length) {
 			await getBackend(ctx).publish(ctx, logs);
+			if (LOKI_HOST && LOKI_WRITE_PCT > randomPct()) {
+				try {
+					await loki.publish(ctx, logs);
+				} catch (err) {
+					captureException(err, 'Failed to publish log to Loki');
+				}
+			}
```

Comment:
Having seen this logic is duplicated in two places I think it'd be worth creating a helper function that does
```ts
const lokiPublish = async (ctx, logs) =>
	if (LOKI_HOST && LOKI_WRITE_PCT > randomPct()) {
		try {
			await loki.publish(ctx, logs);
		} catch (err) {
			captureException(err, 'Failed to publish log to Loki');
		}
	}
```
which will make the calling sites much simpler and also avoid the need to juggle promises as much, eg this site could be simplified to
```ts
await Promise.all([
	getBackend(ctx).publish(ctx, logs),
	lokiPublish(ctx, logs),
]);

Reply:
Moved conditional logic to function, and implemented `Promise.all`